Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportEntitiesDataMapping.cs

```diff
@@ -477,136 +549,295 @@ public static Amdocs.Ginger.CoreNET.Execution.eRunStatus GetRunnerStatus(GingerE
+                        
 
-            accountReportRunner.ChildExecutedItemsCount.Add(new AccountReport.Contracts.Helpers.DictObject
-            { Key = Actions, Value = ChildExecutedItemsCountAction });
+            foreach (BusinessFlow businessFlow in ((GingerExecutionEngine)gingerRunner.Executor).BusinessFlows)
+            {
+                if (businessFlow.Active)
+                {                   
+                    var bfStat = accountReportStatistics.Where(x => x.EntityId == businessFlow.Guid).FirstOrDefault();
+                    if (bfStat!=null)
+                    {                             
+                            accountReportRunner.ChildsExecutionStatistics.Activity.TotalExecutable =
+                                   accountReportRunner.ChildsExecutionStatistics.Activity.TotalExecutable + bfStat.ChildsExecutionStatistics.Activity.TotalExecutable;
+
+                            accountReportRunner.ChildsExecutionStatistics.Activity.TotalExecuted =
+                                   accountReportRunner.ChildsExecutionStatistics.Activity.TotalExecuted + bfStat.ChildsExecutionStatistics.Activity.TotalExecuted;
+
+                            accountReportRunner.ChildsExecutionStatistics.Activity.TotalPassed =
+                               accountReportRunner.ChildsExecutionStatistics.Activity.TotalPassed + bfStat.ChildsExecutionStatistics.Activity.TotalPassed;
+
+                            foreach (var dicObj in bfStat.ChildsExecutionStatistics.Activity.StatusCount)
+                            {
+                                if (accountReportRunner.ChildsExecutionStatistics.Activity.StatusCount.ContainsKey(dicObj.Key))
+                                {
+                                    accountReportRunner.ChildsExecutionStatistics.Activity.StatusCount[dicObj.Key] = accountReportRunner.ChildsExecutionStatistics.Activity.StatusCount[dicObj.Key] + dicObj.Value;
+                                }
+                                else
+                                {
+                                    accountReportRunner.ChildsExecutionStatistics.Activity.StatusCount[dicObj.Key] = dicObj.Value;
+                                }
+                            }                        
+                    }
 
-            accountReportRunner.ChildPassedItemsCount.Add(new AccountReport.Contracts.Helpers.DictObject
-            { Key = Actions, Value = ChildPassedItemsCountAction });
+                    foreach (Activity activity in businessFlow.Activities)
+                    {
+                        if(activity.Active)
+                        {
+                            var stat = accountReportStatistics.Where(x => x.EntityId == activity.Guid).FirstOrDefault();
```

Comment:
Use FirstOrDefault instead of where directly

Reply:
Done.